backend/src/API/devPortfolioAPI.ts

```diff
@@ -0,0 +1,46 @@
+import DevPortfolioDao from '../dao/DevPortfolioDao';
+import PermissionsManager from '../utils/permissionsManager';
+import { PermissionError, BadRequestError } from '../utils/errors';
+import validateSubmission from '../utils/githubUtil';
+
+export const getAllDevPortfolios = async (): Promise<DevPortfolio[]> =>
+  DevPortfolioDao.getAllInstances();
+
+export const createNewDevPortfolio = async (
+  instance: DevPortfolio,
+  user: IdolMember
+): Promise<void> => {
+  const canCreateDevPortfolio = await PermissionsManager.isLeadOrAdmin(user);
+  if (!canCreateDevPortfolio) {
+    throw new PermissionError(
+      `User with email: ${user.email} does not have permission to create dev portfolio!`
+    );
+  }
+  await DevPortfolioDao.createNewInstance(instance);
+};
+
+export const deleteDevPortfolio = async (uuid: string, user: IdolMember): Promise<void> => {
+  const canDeleteDevPortfolio = await PermissionsManager.isLeadOrAdmin(user);
+  if (!canDeleteDevPortfolio) {
+    throw new PermissionError(
+      `User with email: ${user.email} does not have permission to delete dev portfolio!`
+    );
+  }
+  await DevPortfolioDao.deleteInstance(uuid);
+};
+
+export const makeDevPortfolioSubmission = async (
+  uuid: string,
+  submission: DevPortfolioSubmission
+): Promise<void> => {
+  const devPortfolio = DevPortfolioDao.getInstance(uuid) as DevPortfolio;
+  if (!devPortfolio) throw new BadRequestError(`Dev portfolio with uuid ${uuid} does not exist.`);
+
+  if (devPortfolio.earliestDate <= Date.now() && Date.now() <= devPortfolio.deadline) {
+    return DevPortfolioDao.makeDevPortfolioSubmission(
+      uuid,
+      await validateSubmission(devPortfolio, submission)
+    );
+  }
```

Comment:
I think throwing an error might be better here since it's different from the PRs in the submission being all invalid